framework/libs/ktest/src/runner.rs

```diff
@@ -14,6 +14,10 @@ use crate::{
     CatchUnwindImpl, KtestError, KtestItem, KtestIter,
 };
 
+/// Define a function `__ktest_init` to be called before running the tests.
+#[no_mangle]
+fn __ktest_init() {}
```

Comment:
I don't get it. Why will this `__ktest_init` be overridden by the one defined by user? The PR does not take any special measures to mark this function a "weak" or "default" version. How can the compiler be smart enough to choose the user-defined `__ktest_init` over this one?

Reply:
Can anyone be kind enough to shed some lights on this question for me? Thanks.

Reply:
I don't find a clear explaination yet. The compiler asks me to remove `#[linkage = "weak"]` when I add it.  The description in page 195 of [Rust for Rustaceans](https://rust-for-rustaceans.com/) may explain it to some extent, because I only **define**  `__ktest_init` once.
> A quick note about terminology: a symbol can be declared multiple times but defined only once. Every declaration of a symbol will link to the same single definition for that symbol at linking time. If no definition for a declaration is found, or if there are multiple definitions, the linker will complain.

Reply:
> because I only define __ktest_init once.

Isn't the `__ktest_init` function also defined by the target crate of `cargo osdk ktest`? So it is defined twice.